content/academy/anti_scraping/techniques/rate_limiting.md

```diff
@@ -0,0 +1,59 @@
+
+In cases when a higher number of requests is expected for the crawler, using a [proxy]({{@link anti_scraping/proxies.md}}) and rotating the IPs is essential to let the crawler run as smoothly as possible and avoid being blocked.
+
+## [](#dealing-with-rate-limiting) Dealing rate limiting with proxy/session rotating
+
+The most popular and effective way of avoiding rate-limiting issues is by rotating [proxies]({{@link anti_scraping/proxies.md}}) after every **n** number of requests, which makes your scraper appear as if it is making requests from various different places. Since the majority of rate-limiting solutions are based on IP addresses, rotating IPs allows a scraper to make large amounts to a website without getting restricted.
+
+In the Apify SDK, proxies are automatically rotated for you when you use `proxyConfiguration` and a [**SessionPool**]((https://sdk.apify.com/docs/api/session-pool)) within a crawler. The SessionPool handles a lot of the nitty gritty of proxy rotating, especially with [browser based crawlers]({{@link puppeteer_playwright.md}}) by retiring a browser instance after a certain number of requests have been sent from it in order to use a new proxy (a browser instance must be retired in order to use a new proxy).
+
+Here is an example of these features being used in a **PuppeteerCrawler** instance:
+
+```JavaScript
+import Apify from 'apify';
+
+const myCrawler = new Apify.PuppeteerCrawler({
+    proxyConfiguration: await Apify.createProxyConfiguration({
+        groups: ['RESIDENTIAL'],
+    }),
+    sessionPoolOptions: {
+        // Note that a proxy is tied to a session
+        sessionOptions: {
+            // Let's say the website starts blocking requests after
+            // 20 requests have been sent in the span of 1 minute from
+            // a single user.
+            // We can stay on the safe side and retire the browser
+            // and rotate proxies after 15 pages (requests) have been opened.
+            maxUsageCount: 15,
+        },
+    },
+    // ...
+});
+```
+
+> Take a look at the [**Using proxies**]({{@link anti_scraping/proxies/using_proxies.md}}) lesson to learn more about how to use proxies and rotate them in the Apify SDK.
+
+### [](#configuring-session-pool) Configuring a session pool
+
+There are various configuration options available in `sessionPoolOptions` that can be used to set up the SessionPool for different rate-limiting scenarios. In the example above, we used `maxUsageCount` within `sessionOptions` to prevent more than 15 requests from being sent using a session before it was thrown away; however, a maximum age can also be set using `maxAgeSecs`.
+
+When dealing with frequent and unpredictable blockage, the `maxErrorScore` option can be set to trash a session after it's hit a certain number of errors.
+
+To learn more about all configurations available in `sessionPoolOptions`, refer to the [SDK documentation](https://sdk.apify.com/docs/typedefs/session-pool-options).
```

Comment:
Again, let's mention that the default settings are usually good enough so people don't get stuck on the configurations